deviations/templates/deviations/_remove_modal.html

```diff
@@ -0,0 +1,64 @@
+{% load i18n %}
+{% load course %}
+
+<div class="modal" id="remove-selected-modal" tabindex="-1" role="dialog" aria-labelledby="remove-selected-modal-label">
+	<div class="modal-dialog" role="document">
+		<div class="modal-content">
+			<div class="modal-header">
+				<button type="button" class="close" data-dismiss="modal" aria-label="{% translate "CLOSE" %}">
+					<span aria-hidden="true">&times;</span>
+				</button>
+				<h4 class="modal-title" id="remove-selected-modal-label">
+					{% translate "REMOVE_SELECTED_DEVIATIONS" %}
+				</h4>
+			</div>
+
+			<div class="modal-body">
+				<p id="remove-selected-hidden-warning" class="alert alert-warning">
+					{% translate "SOME_DEVIATIONS_ARE_HIDDEN_WARNING" %}
+				</p>
+				<p id="remove-selected-no-selection-warning" class="alert alert-warning">
+					{% translate "NO_DEVIATIONS_SELECTED_WARNING" %}
+				</p>
+				<p id="remove-selected-confirmation"></p>
+			</div>
+
+			<div class="modal-footer">
+				<input
+					class="aplus-button--secondary aplus-button--md"
+					type="button"
+					value="{% translate 'CANCEL' %}"
+					data-dismiss="modal"
+				/>
+				<input
+					id="remove-selected-submit-button"
+					class="aplus-button--danger aplus-button--md"
+					type="submit"
+					value="{% translate 'REMOVE' %}"
+					formaction="{{ remove_url }}"
+				/>
+			</div>
+		</div>
+	</div>
+</div>
+
+<script>
+	$(function() {
+		const modal = $("#remove-selected-modal");
+		const form = modal.closest("form");
+		modal.on('shown.bs.modal', function() {
+			const hiddenSelected = form.find('tr.hidden-filter input[name="id"]:checked');
+			$("#remove-selected-hidden-warning").toggleClass("hidden", hiddenSelected.length === 0);
+
+			const selected = form.find('input[name="id"]:checked');
+			$("#remove-selected-no-selection-warning").toggleClass("hidden", selected.length > 0);
+			$("#remove-selected-confirmation")
+			.toggleClass("hidden", selected.length === 0)
+			.text(
+				_("Are you sure you want to remove the selected deviation? |||| Are you sure you want to remove the %{smart_count} selected deviations?",
+				{ smart_count: selected.length })
+			);
+			$("#remove-selected-submit-button").attr("disabled", selected.length === 0);
```

Comment:
This looks suspicious. The `disabled` attribute is a boolean attribute and it does not use any value. If you set a value, the element is still disabled even if the value is "false" or similar.

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled

Should this be treated like a property?
https://api.jquery.com/prop/

Reply:
It seems that `.attr("disabled", true)` adds an attribute like `disabled="disabled"` and `.attr("disabled", false)` removes the attribute. `prop` works the same way except the attribute it adds looks like `disabled=""`. So it works either way, however, the jQuery documentation states that

> To retrieve and change DOM properties such as the checked, selected, or disabled state of form elements, use the .prop() method.

so I guess I'll change it to use `prop`.